libs/util/http.client.js

```diff
@@ -146,12 +143,14 @@ function doXhr(method, url, headers, data, config, callback) {
                 callback(NULL, response);
             },
             failure : function (err, response) {
-                if (!shouldRetry(method, config, response.statusCode)) {
+                if (!shouldRetry(method, config, response.statusCode, attempt)) {
                     callback(err);
                 } else {
                     setTimeout(
-                        function retryXHR() { doXhr(method, url, headers, data, config, callback); },
-                        config.retry.interval
+                        function retryXHR() {
+                            doXhr(method, url, headers, data, config, attempt + 1, callback);
+                        },
+                        config.retry.interval * Math.pow(2, attempt)
```

Comment:
@pablopalacios since you are at this, do you mind improving this exponential retry logic to use full jitter, as documented [here](https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/) by aws?

Reply:
What should we use as cap? Or should we not use cap at all?

Reply:
@lingyan I updated it but I haven't set `cap` for now so the delay will always be a random number between 0 and `                        config.retry.interval * Math.pow(2, attempt)`. My feeling was that setting a `cap` as in the aws article would require us to set another property in the configuration to allow users to tweak it properly if they have different interval configurations. We could instead, in the future, add another configuration where users can pass a custom function to calculate the delay.

Reply:
Sure!  I am ok with not using cap.  We can add it later if needed.  Thanks @pablopalacios for making the change!